api/pkg/s3/uploadpart.go

```diff
@@ -115,6 +127,16 @@ func (s *APIService) UploadPart(request *restful.Request, response *restful.Resp
 		return
 	}
 
+	// Check if sha256sum match.
+	if needCheckSha256 {
+		sha256Sum := hex.EncodeToString(sha256Writer.Sum(nil))
+		if inputSh256Sum != sha256Sum {
```

Comment:
Just trying to understand, if it's signed payload, this will always be the case right? How is the behavior. Please put comment

Reply:
See response above please.